Add Sprite changes to Logic Gates to show the input/output state

0tito opened the pull request:
## About the PR
Gates now show the state of their inputs and outputs on the sprite

## Why / Balance
Its easier to use them if you can see the states


## Media
https://github.com/user-attachments/assets/c305a2b9-adcb-420f-8a2e-f5ad60a11166

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->


**Changelog**
<!--
:cl:
- tweak: Logic gates now show the state of input/output on their sprite!
-->

deltanedas commented:
use separate layers for a/b/output and none of the evil stuff is needed

0tito commented:
Is this better? im not exactly the best coder :godo:

Aeshus reviewed (changes requested):
Thanks for the PR!

There's some small code style issues, and a bug where the code accidentally sets InputA instead of InputB.

Other than that, I like the change and the rest of the visualizer/layering stuff looks good (I don't have personal experience with any of it).

Also, I'm not a maintainer, so this is just a peer review. I may have made some mistakes in my review, or have missed things, so take this review with a grain of salt.

deltanedas commented on Content.Shared/DeviceLinking/SharedLogicGate.cs:
```diff
@@ -32,5 +35,16 @@ public enum LogicGateVisuals : byte
+}
+
+[Serializable, NetSerializable]
+public enum LogicGateState : byte
```
copy paste of SignalState

Aeshus reviewed (approved):
LGTM! Thanks for applying the changes/feedback.

I think the only thing now is that you could add yourself to the `meta.json`'s copyright to say that you created the sprites for the light-up states.

Also, make sure to click "resolve" for comments/requested changes once you address it. Otherwise, maintainers may have the false impression that it is still a work-in-progress or ongoing issue.

slarticodefast commented on Content.Server/DeviceLinking/Systems/LogicGateSystem.cs:
```diff
@@ -102,10 +102,12 @@ private void OnSignalReceived(EntityUid uid, LogicGateComponent comp, ref Signal
         if (args.Port == comp.InputPortA)
         {
             comp.StateA = state;
+            _appearance.SetData(uid, LogicGateVisuals.InputA, state == SignalState.High ? "High" : "Low");
```
Instead of using the strings "High" and "Low" you can simply make it a boolean

slarticodefast commented on Resources/Prototypes/Entities/Structures/gates.yml:
```diff
@@ -36,6 +36,12 @@
     layers:
     - state: base
     - state: logic
+    - state: empty
```
instead of using an empty sprite just toggle the layers on or off

slarticodefast reviewed (changes requested):
Cool idea. Needs a little more cleanup before it can be merged.
Thanks to Aeshus and delta for doing the previous reviews.

slarticodefast commented:
Quick question: What changes did you make to logic.png? I don't see any difference yet somehow it still doubled in size.

0tito commented:
I used it as base when i was making the other sprites, and i think i changed its sprite when  i was learning how to change sprites but then instead of reverting it i painted it back to a similar color :godo: i used krita, maybe when i saved i changed something on it?

slarticodefast commented:
If it is not changed leave it as it was to preserve the git history and attributions.
And I gave the PR a quick in-game test and the sprites for gates other than the or gate seem to be wrong now when spawning them.
![grafik](https://github.com/user-attachments/assets/9beb7f25-e09d-4266-b8fb-8d4a789c25c1)

0tito commented:
This happens in normal wizden too, iirc oasis has some gates like that, but if you craft them and  switch through them normally they work. I can try to change it tho, also how do i "revert" the change to logic.png, would just replacing the sprite with upstream logic.png fix it? or do i have to do something else

slarticodefast commented:
Just copying the original sprite back is probably the easiest way. There should be a git command, too, but no idea which one.
Thanks for fixing the other gates! I'll give it another test later.

slarticodefast reviewed (approved):
I tested this and it works as intended for non-pulsed signals. With signal buttons there seems to be some weird behaviour for gates, but that is not the fault of this PR and happening on master as well. I opened an issue for that #33716.
Thank you for your first contribution!
